### DISCUSSION TITLE
Edit Views

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#7254] [Questions] [answered]
Views are really awesome! I don't see any way to view, edit, rename or delete the sql for a view. I found a way to delete the view here: https://database.dev/drop-view

But how do I view a view or modify one?

Thanks!

**Comment by silentworks** [comment, marked as answer]
Here are the commands below to do each of what you have requested

### View a VIEW
```sql
SELECT * FROM view_name
```
### Rename a VIEW
```sql
ALTER VIEW view_name RENAME TO new_view_name;
```
### Edit a VIEW
```sql
CREATE OR REPLACE VIEW view_name AS
  new_query
```
### Delete a VIEW
```sql
DROP VIEW IF EXISTS view_name;
```

**Reply by jtkeyva** [reply]
Thanks! Is there any way to see the contents of the view ex: how it was constructed? It would be nice to recall how the view was constructed showing JOINS etc.

**Reply by silentworks** [reply]
You can run this query to get that info
```sql
SELECT pg_get_viewdef('"public"."view_name"'::regclass, true) AS create_statement;
```

**Reply by jtkeyva** [reply]
Awesome thanks!

**Reply by Ivaylo-Korakov** [reply]
So basically people are paying for your product and views are an essential part of every database but you decided not to spend the money time and resources to provide a UI for people to manage their views. Seems like a wise decision for a paid SaaS product. Keep it up :)

**Reply by iveshenry18** [reply]
+1 to the call for an improved View management UI in Supabase Studio.

Views are such a powerful way to manage your data and design your backend to be "Postgres-native". I think better View management capabilities in Supabase Studio are well-aligned with the project's stated goals.

Consider getting this on the roadmap!

**Reply by roland-v9** [reply]
I agree. It took me a few days to work out views created in the dashboard default to bypassing row level security. Exposing egg on face. The docs dont mention either if realtime works with views either, instead focusing on single table filter examples. Views are great. DBA's love them.

**Comment by glatocha** [comment]
![image](https://github.com/supabase/supabase/assets/51866783/dcea2c07-0611-47f3-80c9-8ab97704327b)

bottom right there is a definition data switch

**Reply by parkernilson** [reply]
Oh wow, yeah there it is. I would never have found that if not for you!

**Reply by JKarlavige** [reply]
Thanks @glatocha, exactly what I was looking for!

**Comment by rngadam** [comment]
I was also surprised to find out there's not view editors and apparently that is a conscious decision in this locked and closed issue:

https://github.com/supabase/supabase/issues/2393

_

> We've discussed this a few times and I think we ultimately figured that it would not make sense to have a GUI for editing/creating a view, but rather you write SQL for it.

_
